File: 202105181609 - scurodark1 (CSharp - Rent A Car Automation)/01_source-code/05_project/project/AracKiralamaOtomasyon/Form_Musteri.cs
using System;
using System.Collections;
using System.Windows.Forms;

namespace AracKiralamaOtomasyon
{
    public partial class Form_Musteri : Form
    {
        Class_Islemler islemler = new Class_Islemler();
        string tablo = "musteri";
        int Id = 0;
        public Form_Musteri()
        {
            InitializeComponent();
        }
        public Form_Musteri(int _Id)
        {
            InitializeComponent();
            Id = _Id;
        }

        private void Form_Personel_Load(object sender, EventArgs e)
        {
            if (Id != 0)
            {
                try
                {
                    ArrayList veriler = islemler.Getir(tablo, Id);

                    txtTcno.Text = veriler[1].ToString();
                    txtAd.Text = veriler[2].ToString();
                    txtSoyad.Text = veriler[3].ToString();
                    txtTelefon.Text = veriler[4].ToString();
                    txtEposta.Text = veriler[5].ToString();
                    txtAdres.Text = veriler[6].ToString();

                    this.Text += " Güncelle";
                    btn_Ekle.Text = "Güncelle";
                    btn_Sil.Visible = true;
                }
                catch
                {
                    return;
                }
            }
        }

        private void btn_Ekle_Click(object sender, EventArgs e)
        {
            ArrayList kayit = new ArrayList()
            {
                new ArrayList(){
                    "tcno",txtTcno.Text
                },
                   new ArrayList(){
                    "ad",txtAd.Text
                },
                      new ArrayList(){
                    "soyad",txtSoyad.Text
                },

               new ArrayList(){
                    "telefon",txtTelefon.Text
                },
                new ArrayList(){
                    "eposta",txtEposta.Text
                },

               new ArrayList(){
                    "adres",txtAdres.Text
                },

            };

            if (Id != 0)
            {
                kayit.Add(new ArrayList() { "Id", Id });
                if (islemler.Guncelle(tablo, kayit))
                    islemler.MesajKutu(tablo + " güncelleme");
                else
                    islemler.MesajKutu(2, tablo + " güncelleme");
                return;
            }
            if (islemler.Ekle(tablo, kayit))
                islemler.MesajKutu(tablo + " ekleme");
            else
                islemler.MesajKutu(2, tablo + " ekleme");
            islemler.Temizle(this);
        }

        private void btn_Sil_Click(object sender, EventArgs e)
        {
            islemler.Sil(this, tablo, Id);
        }
    }
}

File: 202101240037 - Kielmoritz (VbNet - Cargo Automation)/01_source-code/05_project/VBOtomasyon/VBOtomasyon/Form_Musteri.vb
Public Class Form_Musteri
    Dim tablo As String = "Musteri_Tablosu_F"
    Public Id As Integer = 0

    Dim islemler As Class_Islemler = New Class_Islemler
    Private Sub Form_Musteri_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If Id <> 0 Then
            Getir()
        End If
    End Sub

    Public Sub Getir()
        btnEkle.Text = "GÜNCELLE"
        btnSil.Visible = True
        Dim veriler As ArrayList = islemler.Getir(tablo, Id)
        txtTc.Text = veriler(1).ToString
        txtAd.Text = veriler(2).ToString
        txtSoyad.Text = veriler(3).ToString
        txtAdres.Text = veriler(4).ToString
    End Sub

    Private Sub btnEkle_Click(sender As Object, e As EventArgs) Handles btnEkle.Click
        Dim veriler As ArrayList = New ArrayList

        veriler.Add(New ArrayList() From {"TC_KimlikNo", txtTc.Text})
        veriler.Add(New ArrayList() From {"Adi", txtAd.Text})
        veriler.Add(New ArrayList() From {"Soyadi", txtSoyad.Text})
        veriler.Add(New ArrayList() From {"Adres", txtAdres.Text})

        If Id <> 0 Then
            veriler.Add(New ArrayList() From {"Id", Id})
            If islemler.Guncelle(tablo, veriler) Then
                islemler.MesajKutu(tablo + " güncelleme")
            Else
                islemler.MesajKutu(2, tablo + " güncelleme")
            End If
            Return
        End If

        If islemler.Ekle(tablo, veriler) Then
            islemler.MesajKutu(tablo + " ekleme")
        Else
            islemler.MesajKutu(2, tablo + " ekleme")
        End If

    End Sub

    Private Sub btnSil_Click(sender As Object, e As EventArgs) Handles btnSil.Click
        islemler.Sil(Me, tablo, Id)
    End Sub
End Class